proto/sunrise/swap/metadata.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+
+package sunrise.swap;
+
+import "amino/amino.proto";
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+import "sunrise/swap/route.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/sunriselayer/sunrise/x/swap/types";
+
+message PacketMetadata {
+  SwapMetadata swap = 1;
+}
+
+message ForwardMetadata {
+  string receiver = 1;
+  string port = 2;
+  string channel = 3;
+  google.protobuf.Duration timeout = 4 [
+    (gogoproto.nullable) = false, (gogoproto.stdduration) = true
+  ];
+  uint32 retries = 5; 
+
+  // Using Google's Any to handle the flexible JSON object
+  google.protobuf.Any next = 6;
+}
+
+message ExactAmountIn {
+  string min_amount_out = 1 [
+    (cosmos_proto.scalar)  = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable)   = false,
+    (amino.dont_omitempty) = true
+  ];
+}
+
+message ExactAmountOut {
+  string amount_out = 1 [
+    (cosmos_proto.scalar)  = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable)   = false,
+    (amino.dont_omitempty) = true
+  ];
+  ForwardMetadata change = 2;
+}
+
+message SwapMetadata {
+  string interface_provider = 1;
+  Route route = 2;
+  oneof swap_type {
```

Comment:
Can you rename to `amount_strategy`?

Reply:
amount?? why?

Reply:
The word strategy is a little unnatural, but it's better than swap_type, i think.
mode or type is less unnatural.